### PR TITLE
Add Image Search API Swagger

### PR DESCRIPTION
**Pull request by shakao**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json [hunk 1]
```diff
@@ -0,0 +1,2171 @@
...
+          "Successful image search": {
+            "$ref": "./examples/SuccessfulImageSearchRequest.json"
+          }
+        }
+      }
+    },
+    "/images/details": {
+      "get": {
+        "summary": "The Image Detail Search API lets you search on Bing and get back insights about an image, such as webpages that include the image. This section provides technical details about the query parameters and headers that you use to request insights of images and the JSON response objects that contain them. For examples that show how to make requests, see [Searching the Web for Images](https://docs.microsoft.com/azure/cognitive-services/bing-image-search/search-the-web).",
+        "operationId": "Imagess_Details",
```
Comment (alvadb) [inline specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: Please remove the double "s" on Imagess.

File: specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json [hunk 2]
```diff
@@ -0,0 +1,2171 @@
...
+            "in": "header",
+            "description": "The IPv4 or IPv6 address of the client device. The IP address is used to discover the user's location. Bing uses the location information to determine safe search behavior. Although optional, you are encouraged to always specify this header and the X-Search-Location header. Do not obfuscate the address (for example, by changing the last octet to 0). Obfuscating the address results in the location not being anywhere near the device's actual location, which may result in Bing serving erroneous results.",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "X-Search-Location",
+            "x-ms-client-name": "Location",
+            "in": "header",
+            "description": "A semicolon-delimited list of ke
```
Comment (alvadb) [inline specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: Please add missing property descriptions.

Comment (alvadb) [inline specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: Should this be readOnly?

Comment (alvadb) [inline specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: Please add description to all properties.

Comment (alvadb) [inline specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: Please add description to properties.

Comment (alvadb) [inline specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: Should this be readOnly?

Comment (alvadb) [inline specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: Please add description.

Comment (alvadb) [inline specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: Please add descriptions to the properties.

Comment (alvadb) [inline specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: Please add a description.

**Review by alvadb — CHANGES_REQUESTED** [review]
There are many properties that are missing description and possible readOnly tags.  Please add them.

**Comment by alvadb** [comment]
Please review and fix linter and model errors.
Linter: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/309660037
Model: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/309660038

File: specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json (hunk 1 above)
Comment (shakao) [reply specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: resolved

File: specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json (hunk 2 above)
Comment (shakao) [reply specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: resolved

Comment (shakao) [reply specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: resolved

Comment (shakao) [reply specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: resolved

Comment (shakao) [reply specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: resolved

Comment (shakao) [reply specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: resolved

Comment (shakao) [reply specification/cognitiveservices/data-plane/ImageSearch/v1.0/ImageSearch.json]: resolved

**Review by alvadb — APPROVED** [review]

**Pull request merged**